lib/jsdom/browser/xhr/xmlhttprequest.js

```diff
@@ -0,0 +1,891 @@
+"use strict";
+
+var HTTP_STATUS_CODES = require("http").STATUS_CODES;
+var deasync;
+try {
+  deasync = require("deasync");
```

Comment:
Augh :(. We cannot use deasync, because it is a native module, and those cause a lot of pain.

Could you instead investigate the techniques in https://github.com/ForbesLindesay/sync-request ?